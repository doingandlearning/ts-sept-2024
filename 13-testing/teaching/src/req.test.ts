import fakeFetch from "./req-promise";
import { test, expect, vi, beforeEach, afterEach } from "vitest";

beforeEach(() => {
  vi.useFakeTimers();
});

afterEach(() => {
  vi.useRealTimers();
});

test("responds with data", async () => {
  const responsePromise = fakeFetch("https://bbc.com");
  vi.runAllTimers();
  const response = await responsePromise;
  expect(Buffer.isBuffer(response)).toBe(true);
  expect(response.toString()).toMatchInlineSnapshot(`"some other data!!!"`);
});

test("unhappy path for fakeFetch", async () => {
  const responsePromise = fakeFetch("http://error.com");
  vi.runAllTimers();
  await expect(responsePromise).rejects.toThrow("network error");

  try {
    const responsePromise = fakeFetch("http://error.com");
    vi.runAllTimers();
    await expect(responsePromise).rejects.toThrow("network error");
    const response = await responsePromise;
  } catch (error) {
    expect(error).toBeDefined();
    expect(error).toMatchInlineSnapshot(`[Error: network error]`);
  }
});
