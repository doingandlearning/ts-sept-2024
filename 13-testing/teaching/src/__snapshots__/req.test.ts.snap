// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`responds with data 1`] = `"some other data!!!"`;
